src/middlewared/middlewared/plugins/system/dmi.py

```diff
@@ -18,18 +21,17 @@ class SystemService(Service):
         'has-ipmi': None,
     }
 
-    @private
-    def dmidecode_info(self):
-        if all(v is None for k, v in SystemService.CACHE.items()):
+    def info(self):
```

Comment:
Since this is a standard class outside of middleware, we should probably just decorate this function with `functools.cached_property` so we don't have to do the whole `self.CACHE` dance. Do note, however, we'll probably have to look at the unit tests that I wrote because I'm not sure how doing that will affect those off the top of my head.

Reply:
Yes, but what's the point of fixing something that already works?